profitbricks/src/main/java/org/jclouds/profitbricks/domain/Image.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.domain;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class Image {
+
+   public enum Type {
+
+      HDD, CDROM, UNRECOGNIZED;
+
+      public static Type fromValue(String v) {
+         try {
+            return valueOf(v);
+         } catch (IllegalArgumentException ex) {
+            return UNRECOGNIZED;
+         }
+      }
+   }
+
+   public abstract String id();
+
+   public abstract String name();
+
+   public abstract Float size(); // MB
+
+   public abstract Type type();
+
+   public abstract Location location();
+
+   public abstract OsType osType();
+
+   public abstract boolean isPublic();
+
+   public abstract boolean isWriteable();
+
+   public abstract boolean isBootable();
+
+   public abstract boolean isCpuHotPlug();
+
+   public abstract boolean isCpuHotUnPlug();
+
+   public abstract boolean isRamHotPlug();
+
+   public abstract boolean isRamHotUnPlug();
+
+   public abstract boolean isNicHotPlug();
+
+   public abstract boolean isNicHotUnPlug();
+
+   public abstract boolean isDiscVirtioHotPlug();
+
+   public abstract boolean isDiscVirtioHotUnPlug();
```

Comment:
So far, the [API](https://www.profitbricks.com/apidoc/1_3/APIDocumentation.html?GetAllImages.html) is always returning values for those properties and it's docs don't say that they're nullable. I think it's safe to assume that they're not.